validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -857,8 +857,15 @@ def get_transaction(self, index):
 
     def cancel(self):
         with self._condition:
-            self._cancelled = True
-            self._condition.notify_all()
+            if not self._cancelled and not self._final:
+                contexts = [tr.context_id for tr in self._txn_results.values()
+                            if tr.context_id]
+                self._squash(self._first_state_hash,
```

Comment:
Why do the contexts need to be squashed?  Can't the just be removed?  Seems like a lot of unnecessary computation to just clean up the contexts.

Reply:
Granted, it fixes the leak, and I can address the performance implications while working on [STL-778](https://jira.hyperledger.org/browse/STL-778)

Reply:
There would have to be a second method from the context manager that the schedulers would take that would delete contexts.